config/utils.ts

```diff
@@ -0,0 +1,447 @@
+    );
+  }
+
+  const accountId = parseInt(accountIdVar);
+  const httpTimeout = httpTimeoutVar ? parseInt(httpTimeoutVar) : undefined;
+  const httpUseLocalhost = httpUseLocalhostVar
+    ? httpUseLocalhostVar === 'true'
+    : undefined;
+  const allowUsageTracking = allowUsageTrackingVar
+    ? allowUsageTrackingVar === 'true'
+    : undefined;
+  const defaultCmsPublishMode =
+    defaultCmsPublishModeVar === CMS_PUBLISH_MODE.draft ||
+    defaultCmsPublishModeVar === CMS_PUBLISH_MODE.publish
+      ? defaultCmsPublishModeVar
+      : undefined;
+
+  const env = getValidEnv(hubspotEnvironment);
+
+  let account: HubSpotConfigAccount;
+
+  if (personalAccessKey) {
+    account = {
+      authType: PERSONAL_ACCESS_KEY_AUTH_METHOD.value,
+      accountId,
+      personalAccessKey,
+      env,
+      name: accountIdVar,
+      auth: {
+        tokenInfo: {},
+      },
+    };
+  } else if (clientId && clientSecret && refreshToken) {
+    account = {
+      authType: OAUTH_AUTH_METHOD.value,
+      accountId,
+      auth: {
+        clientId,
+        clientSecret,
+        scopes: OAUTH_SCOPES.map((scope: { value: string }) => scope.value),
+        tokenInfo: {
+          refreshToken,
+        },
+      },
+      env,
+      name: accountIdVar,
+    };
+  } else if (apiKey) {
+    account = {
+      authType: API_KEY_AUTH_METHOD.value,
+      accountId,
+      apiKey,
+      env,
+      name: accountIdVar,
+    };
+  } else {
+    throw new Error(
+      i18n('config.utils.buildConfigFromEnvironment.invalidAuthType')
+    );
+  }
+
+  return {
+    accounts: [account],
+    defaultAccount: accountId,
+    httpTimeout,
+    httpUseLocalhost,
+    allowUsageTracking,
+    defaultCmsPublishMode,
+  };
+}
+
+export function getAccountIdentifierAndType(
+  accountIdentifier: string | number
+): { identifier: string | number; identifierType: 'name' | 'accountId' } {
+  const identifierAsNumber =
+    typeof accountIdentifier === 'number'
+      ? accountIdentifier
+      : parseInt(accountIdentifier);
+  const isId = !isNaN(identifierAsNumber);
+
+  return {
+    identifier: isId ? identifierAsNumber : accountIdentifier,
+    identifierType: isId ? 'accountId' : 'name',
+  };
+}
+
+export function getConfigAccountByIdentifier(
+  accounts: Array<HubSpotConfigAccount>,
+  identifierFieldName: 'name' | 'accountId',
+  identifier: string | number
+): HubSpotConfigAccount | undefined {
+  return accounts.find(account => account[identifierFieldName] === identifier);
+}
+
+export function getConfigAccountByInferredIdentifier(
+  accounts: Array<HubSpotConfigAccount>,
+  accountIdentifier: string | number
+): HubSpotConfigAccount | undefined {
+  const { identifier, identifierType } =
+    getAccountIdentifierAndType(accountIdentifier);
+  return accounts.find(account => account[identifierType] === identifier);
+}
+
+export function getConfigAccountIndexById(
+  accounts: Array<HubSpotConfigAccount>,
+  id: string | number
```

Comment:
Is it possible for accountId to be anything except for a number in this new format?

Reply:
Good catch, this should just be number